### PR TITLE
Bump Kotlin to 1.8.21

### PR DESCRIPTION
**Pull request by geoand**
See what's new [here](https://kotlinlang.org/docs/whatsnew1820.html).

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -20,8 +20,8 @@
         <!-- These properties are needed in order for them to be resolvable by the generated projects -->
         <!-- Quarkus uses jboss-parent and it comes with 3.8.1-jboss-1, we don't want that in the templates -->
         <compiler-plugin.version>3.11.0</compiler-plugin.version>
-        <kotlin.version>1.8.10</kotlin.version>
-        <dokka.version>1.8.10</dokka.version>
+        <kotlin.version>1.8.20</kotlin.version>
+        <dokka.version>1.8.20</dokka.version>
```
Comment (gsmet) [inline build-parent/pom.xml:24]: Are you sure 1.8.20 of Dokka was already released, it's usually lagging a lot.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Doesn't look good from a CI perspective.

**Comment by geoand** [comment]
Indeed

**Comment by gsmet** [comment]
I rebased and updated to 1.8.21. Let's see how it goes.

**Comment by gsmet** [comment]
The Gradle build is failing because `QuarkusPluginTest#shouldNotFailOnProjectDependenciesWithoutMain` is not passing.

> Cannot access 'org.jetbrains.kotlin.gradle.plugin.KotlinTargetsContainerWithPresets' which is a supertype of 'org.jetbrains.kotlin.gradle.dsl.KotlinTargetContainerWithPresetFunctions'. Check your module classpath for missing or conflicting dependencies

Code is here: https://github.com/quarkusio/quarkus/blob/main/devtools/gradle/gradle-application-plugin/src/test/java/io/quarkus/gradle/QuarkusPluginTest.java#LL108C1-L108C1

I rebased to upgrade to Gradle 8.1.1, let's see how it goes.

It might be a problem of versions of the plugin as I see we inject a Kotlin Gradle dependency without setting the plugin version.

@snazy maybe you have an idea.

**Comment by snazy** [comment]
This is a strange one. I'll have a look.

**Comment by snazy** [comment]
@geoand opened a PR against your branch

**Comment by snazy** [comment]
Saw that the `gradle-model` has an `implementation` dependency to Quarkus' Kotlin version, which could leak to user build scripts, which might use other Kotlin plugin versions. Updated the code a little in #33091

**Comment by geoand** [comment]
Interesting, thanks!

**Comment by gastaldi** [comment]
This needs a rebase before we can merge

**Comment by geoand** [comment]
Looks good to me

**Pull request merged**